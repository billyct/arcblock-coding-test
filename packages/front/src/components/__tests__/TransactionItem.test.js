import {render} from '@testing-library/react'

jest.mock('../TransactionIO', () => 'transaction-io-component')
jest.mock('../CoinbaseInput', () => 'coinbase-input-component')
jest.mock('../OpReturnOutput', () => 'op-return-output-component')
jest.mock('../Hash', () => 'hash-component')

const TransactionItem = require('../TransactionItem').default

afterAll(() => {
  jest.resetModules()
})

describe.each([
  [
    {
      fee: "0",
      hash: "4a5e1e4baab89f3a32518a88c31bc87f618f76673e2cc77ab2127b7afdeda33b",
      inputs: [{isCoinbaseInput: true}],
      inputsCount: 1,
      inputsValueCount: "0",
      outputs: [{addr: "1A1zP1eP5QGefi2DMPTfTL5SLmv7DivfNa", value: "5000000000", spent: false}],
      outputsCount: 1,
      outputsValueCount: "5000000000",
      time: 1231006505,
    }
  ],
  [
    {
      "hash": "9eb415caf7924327507aae440be9b47ea24acdb5c3e8a2d1cbe04430bab2f396",
      "time": 1611217482,
      "inputs": [
        {
          "addr": "19wyxrJWhDsDtkGMBxe3JGJ75rkrx9kKFf",
          "value": "546",
          "spent": true
        },
        {
          "addr": "1WnotWnmDHfihogsjgWS1uzCGbhUeSNZ4",
          "value": "80270672",
          "spent": true
        }
      ],
      "outputsCount": 3,
      "outputsValueCount": "80142458",
      "outputs": [
        {
          "addr": "1WnotWnmDHfihogsjgWS1uzCGbhUeSNZ4",
          "value": "80141912",
          "spent": true
        },
        {
          "isOpReturnOutput": true
        },
        {
          "addr": "1WnotWnmDHfihogsjgWS1uzCGbhUeSNZ4",
          "value": "546",
          "spent": false
        }
      ]
    }
  ],
])('test TransactionItem Component case %#', transaction => {
  it('should render correct', function () {
    const {container} = render(<TransactionItem transaction={transaction}/>)
    expect(container).toMatchSnapshot()
  })
})

