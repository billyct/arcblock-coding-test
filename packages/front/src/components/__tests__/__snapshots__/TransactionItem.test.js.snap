// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test TransactionItem Component case 0 should render correct 1`] = `
<div>
  <div
    class="w-full leading-loose mb-4"
  >
    <div
      class="flex justify-between items-center bg-gray-100 px-6 py-2"
    >
      <hash-component
        classname="text-base flex-1"
      >
        4a5e1e4baab89f3a32518a88c31bc87f618f76673e2cc77ab2127b7afdeda33b
      </hash-component>
      <div>
        Fee: 
        0 BTC
      </div>
    </div>
    <div
      class="flex text-sm"
    >
      <div
        class="flex-1"
      >
        <div
          class="flex justify-between pl-6 pr-10 py-4 font-bold border-b border-gray-200"
        >
          <div>
            Input (
            1
            )
          </div>
          <div>
            0 BTC
          </div>
        </div>
        <div
          class="px-6 py-2"
        >
          <div>
            <coinbase-input-component />
          </div>
        </div>
      </div>
      <div
        class="w-20 flex flex-col justify-center items-center"
      >
        <svg
          class="text-blue-500 bg-blue-50 rounded-full p-1 box-content"
          height="20"
          width="20"
        >
          right.svg
        </svg>
      </div>
      <div
        class="flex-1"
      >
        <div
          class="flex justify-between pl-6 pr-10 py-4 font-bold border-b border-gray-200"
        >
          <div>
            Output (
            1
            )
          </div>
          <div>
            50.00000000 BTC
          </div>
        </div>
        <div
          class="px-6 py-2"
        >
          <div>
            <transaction-io-component
              data="[object Object]"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="italic text-xs text-gray-500 px-6 py-2"
    >
      2009-01-03 18:15:05
    </div>
  </div>
</div>
`;

exports[`test TransactionItem Component case 1 should render correct 1`] = `
<div>
  <div
    class="w-full leading-loose mb-4"
  >
    <div
      class="flex justify-between items-center bg-gray-100 px-6 py-2"
    >
      <hash-component
        classname="text-base flex-1"
      >
        9eb415caf7924327507aae440be9b47ea24acdb5c3e8a2d1cbe04430bab2f396
      </hash-component>
      <div>
        Fee: 
        NaN BTC
      </div>
    </div>
    <div
      class="flex text-sm"
    >
      <div
        class="flex-1"
      >
        <div
          class="flex justify-between pl-6 pr-10 py-4 font-bold border-b border-gray-200"
        >
          <div>
            Input (
            )
          </div>
          <div>
            NaN BTC
          </div>
        </div>
        <div
          class="px-6 py-2"
        >
          <div>
            <transaction-io-component
              data="[object Object]"
            />
          </div>
          <div>
            <transaction-io-component
              data="[object Object]"
            />
          </div>
        </div>
      </div>
      <div
        class="w-20 flex flex-col justify-center items-center"
      >
        <svg
          class="text-blue-500 bg-blue-50 rounded-full p-1 box-content"
          height="20"
          width="20"
        >
          right.svg
        </svg>
      </div>
      <div
        class="flex-1"
      >
        <div
          class="flex justify-between pl-6 pr-10 py-4 font-bold border-b border-gray-200"
        >
          <div>
            Output (
            3
            )
          </div>
          <div>
            0.80142458 BTC
          </div>
        </div>
        <div
          class="px-6 py-2"
        >
          <div>
            <transaction-io-component
              data="[object Object]"
            />
          </div>
          <div>
            <op-return-output-component />
          </div>
          <div>
            <transaction-io-component
              data="[object Object]"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="italic text-xs text-gray-500 px-6 py-2"
    >
      2021-01-21 08:24:42
    </div>
  </div>
</div>
`;
